internal/ingress/store/store.go

```diff
@@ -286,12 +286,15 @@ func (s Store) GetKongPlugin(namespace, name string) (*configurationv1.KongPlugi
 
 // GetKongClusterPlugin returns the 'name' KongClusterPlugin resource.
 func (s Store) GetKongClusterPlugin(name string) (*configurationv1.KongClusterPlugin, error) {
+	if s.stores.ClusterPlugin == nil {
+		return nil, ErrNotFound{fmt.Sprintf("KongClusterPlugins are not enabled")}
+	}
```

Comment:
To streamline execution under the hood, consider leaving store as is (that is, including a ClusterPlugin that is always empty if KongClusterPlugins are not enabled). Prior art exists: [`EmptyStore`](https://github.com/Kong/kubernetes-ingress-controller/blob/84ba4a2d23fa9a0bb8bc332230c6793b828dc4c1/cli/ingress-controller/main.go#L371).

Practically speaking, all clients of store (+ fakestore) that might have potential to be exposed to stores.ClusterPlugin's nullability, would need to test this special case.

Also, making a field nullable is [a billion-dollar mistake](https://hinchman-amanda.medium.com/null-pointer-references-the-billion-dollar-mistake-1e616534d485) in some cases, for example if you get to design Java.